src/Raven.Client/Documents/Operations/MaintenanceOperationExecutor.cs

```diff
@@ -112,5 +145,15 @@ private IDisposable GetContext(out JsonOperationContext context)
                 throw new InvalidOperationException("Cannot use Maintenance without a database defined, did you forget to call ForDatabase?");
             return RequestExecutor.ContextPool.AllocateOperationContext(out context);
         }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private void ApplyNodeTagAndShardNumberToCommandIfSet<T>(RavenCommand<T> command)
+        {
+            if (_nodeTag != null)
+                command.SelectedNodeTag = _nodeTag;
```

Comment:
IIRC, this will not proxy/redirect.. but will send to request directly to the selected node, not sure if this is an issue though.

Reply:
I can live with that for now.